### ISSUE TITLE
Unused api tokens from vocabulary file vocab.apiseq.pkl

### ISSUE THREAD

**Issue by SantoshMadhukar** [guxd/deep-code-search#9]
I tried to check the frequency of apiseq tokens of vocab.apiseq.pkl file from the training (train.apiseq.h5) and test set (test.apiseq.h5). I found only 9869 from 10000 tokens are used more than once. **Is there any proper reason?**

**Comment by guxd** [comment]
The vocabulary size is chosen by trading off between high coverage and small size (complexity).
It is OK to use the full vocabulary, namely, including all API tokens in the vocabulary as well.

**Comment by guxd** [comment]
The tokens in the vocabulary are just summarized from the training data. So shouldn't be dumb tokens. Did you check the complete training dataset in Google Driver?

**Comment by SantoshMadhukar** [comment]
Hi,
I have checked files from [Google driver](https://drive.google.com/drive/folders/1GZYLT_lzhlVczXjD6dgwVUvDDPHMB6L7) and still found the same problem. To get the complete picture i have shared my [code](https://gist.github.com/SantoshMadhukar/7379d94abc548d0cd6a5a404b236ab3c), please have a look at it. The function defination in the shared code is taken from the keras [implementation](https://github.com/guxd/deep-code-search/blob/fdd7cfa48a8e30c602b9a73c76758893e6124752/keras/codesearcher.py#L119)  (file name codesearcher.py) which is a util to load data into the model.

Sorry in advance if i have done any mistake.

**Comment by guxd** [comment]
It might be due to the legacy version of vocabulary. During binarizing, we updated the datasets frequently while keeping using the same vocabulary. Some words in the vocabulary could remain unchanged and are unused in the new dataset. 
However, I do not think this affects the performance significantly. You can reduce the vocabulary by removing those unused words.

**Comment by SantoshMadhukar** [comment]
ok thank you

**Comment by ttbuffey** [comment]
@guxd are the vocab files from Google driver extracted from the 18,233,872 source code?

**Comment by ttbuffey** [comment]
Also we replaced the search codebase as our own repository, which have different vocabulary than yours, should we replace the vocab.**.PKL files to make more precise search?

**Comment by guxd** [comment]
yes, the vocab files are extracted from the provided source code. 
You should create your own vocabulary when using your own repository.

**Comment by ttbuffey** [comment]
If we replace the vocabulary, do we need to retrain the model with this new vocabulary files, and then do embedding and search?

**Comment by guxd** [comment]
of course

**Comment by ttbuffey** [comment]
Hi @guxd,
Also I am wondering if you could help to upload the original training  ***.txt** files of the 18,233,87 functions:
 - train.apiseq.txt, **train.desc.txt**, train.methname.txt, train.token.txt
Because we want to add some our own code into your training data together for another training, so that we could have our own vocabulary for search.

Great thanks for our continuous support.
Best Regards,
ttbuffey

**Comment by guxd** [comment]
why not just decipher them using the vocabulary?

**Comment by ttbuffey** [comment]
It's a good suggestion, let me try.
Thanks.
